dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.43/V2_43_21__split_event_table.sql

```diff
@@ -34,6 +42,16 @@ alter table singleevent alter column status set not null;
 alter table singleevent alter column eventdatavalues set default '{}'::jsonb;
 alter table singleevent alter column eventdatavalues set not null;
 
+-- Make occurreddate column not nullable. Raise an error if null values are present.
+do $$
+    begin
+        alter table singleevent alter column occurreddate set not null;
+    EXCEPTION
+        WHEN not_null_violation THEN
+            RAISE EXCEPTION 'There is inconsistent data in your DB. Please check https://github.com/dhis2/dhis2-releases/blob/master/releases/2.43/migration-notes.md#null-occurred-date-for-single-event to have more information on the issue and to find ways to fix it. Detailed error message: %', SQLERRM;
```

Comment:
this link and the one below don't exist yet, will you create them later on?

Reply:
Yes, we will take care of documentation here https://dhis2.atlassian.net/browse/DHIS2-19614